src/modules/transaction/hooks/useTransactionFee/utils.test.js

```diff
@@ -0,0 +1,98 @@
+import wallets from "@tests/constants/wallets";
+import moduleCommandSchemas from "@tests/constants/schemas";
+import { computeFee } from "./utils";
+
+const transactionBase = {
+  nonce: BigInt(0),
+  senderPublicKey: Buffer.from(wallets.genesis.summary.publicKey, 'hex'),
+};
+const defaultPriorities = [
+  { value: 0, title: 'Low' },
+  { value: 1, title: 'Medium' },
+  { value: 2, title: 'High' },
+];
+
+const bufferify = (string) => Buffer.from(string, 'hex');
+
+describe('computeFee', () => {
+  it('Returns zero if transaction is not valid', () => {
+    expect(computeFee(null, null, null, null, false)).toEqual(BigInt(0));
+  });
+
+  describe('Normal account', () => {
+    const tokenTransfer = {
+      ...transactionBase,
+      module: 'token',
+      command: 'transfer',
+      params: {
+        tokenId: bufferify('00000000'),
+        amount: BigInt('1000000'),
+        recipientAddress: bufferify(wallets.genesis.summary.address),
+        data: '',
+      },
+      signatures: [],
+    };
+    const schema = moduleCommandSchemas["token:transfer"];
+    const auth = {
+      numberOfSignatures: 1,
+    };
+
+    it('Returns the calculated fee given transaction is valid', () => {
+      const priorities = defaultPriorities.map((item) => ({ ...item, selected: item.title === 'Low' }));
+      expect(computeFee(tokenTransfer, schema, auth, priorities, true)).toEqual(BigInt(133000));
+    });
+
+    it('Returns the calculated fee with higher priority given the transaction is valid', () => {
+      const priorities = defaultPriorities.map((item) => ({ ...item, selected: item.title === 'Medium' }));
+      expect(computeFee(tokenTransfer, schema, auth, priorities, true)).toEqual(BigInt(133063));
+    });
+  });
+
+  describe('Register multisignature', () => {
+    const registerMultisignature = {
+      ...transactionBase,
+      module: 'auth',
+      command: 'registerMultisignature',
+      params: {
+        numberOfSignatures: 2,
+        mandatoryKeys: [bufferify(wallets.genesis.summary.publicKey), bufferify(wallets.validator.summary.publicKey)],
+        optionalKeys: [],
+        signatures: [],
+      },
+      signatures: [],
+    };
+    const schema = moduleCommandSchemas["auth:registerMultisignature"];
+    const auth = {
+      numberOfSignatures: 1,
+    };
+
+    it('Returns the calculated fee given transaction is valid', () => {
+      const priorities = defaultPriorities.map((item) => ({ ...item, selected: item.title === 'Low' }));
+      expect(computeFee(registerMultisignature, schema, auth, priorities, true)).toEqual(BigInt(275000));
```

Comment:
instead of hardcoding, we could snapshot these for all types of transaction so that in future when someone accidentally changes the logic the snapshot can capture?

also can you compute for all transactions? so that we can ensure the fee calculation is done correctly for all types of transaction?

Reply:
maybe this will help to compose transactions https://github.com/LiskHQ/lisk-sdk/blob/transaction_fixture/framework/tx/tx_samples.json

Reply:
TBH I never liked the snapshots. The fact that you can run with `-u` and not care how and why the snapshots were updated worries me. I'd rather bring the attention of the developer down to the test code line to at least see what they're about to change.
Though if you insist I can change it to snapshots